internal/service/entity/details.go

```diff
@@ -0,0 +1,133 @@
+package entity
+
+import (
+	"encoding/binary"
+	"fmt"
+	"github.com/dgraph-io/badger/v3"
+	"github.com/mimiro-io/datahub/internal/service/namespace"
+	"github.com/mimiro-io/datahub/internal/service/store"
+	"github.com/mimiro-io/datahub/internal/service/types"
+)
+
+type Lookup struct {
+	badger     store.BadgerStore
+	namespaces namespace.Manager
+}
+
+func NewLookup(s store.BadgerStore) (Lookup, error) {
+	ns := namespace.NewManager(s)
+	return Lookup{s, ns}, nil
+}
+
+func (l Lookup) Details(id string, datasetNames []string) (map[string]interface{}, error) {
+	curie, err := l.asCURIE(id)
+	if err != nil {
+		return nil, err
+	}
+	b := l.badger.GetDB()
+
+	rtxn := b.NewTransaction(false)
+	defer rtxn.Discard()
+	internalId, err := l.internalIdForCURIE(rtxn, curie)
+	if err != nil {
+		return nil, err
+	}
+
+	scope := l.badger.LookupDatasetIDs(datasetNames)
+	details, err := l.loadDetails(rtxn, internalId, scope)
+	if err != nil {
+		return nil, err
+	}
+	return details, nil
+}
+
+func (l Lookup) loadDetails(rtxn *badger.Txn, internalEntityId types.InternalID, scope []types.InternalDatasetID) (map[string]interface{}, error) {
+	result := map[string]interface{}{}
+	/*
+		binary.BigEndian.PutUint16(entityIdBuffer, ENTITY_ID_TO_JSON_INDEX_ID)
```

Comment:
Why are these lines commented out?

Reply:
It's just a hint about the key structure we work with